### PR TITLE
Otto in Sandboxes

### PR DESCRIPTION
**Pull request by clintonb**
TODO
- [x] Ecom service starts, and OAuth setup properly
- [x] Set correct Nginx ports
- [ ] Verify devstack not broken
- [x] Move OAuth setup to either its own role or a new role for all local, post-edxapp, tasks
- ~~Migrate courses~~ (de-prioritized by @clintonb)

ECOM-2185

FYI @feanil @rlucioni @jimabramson @wedaly

### CODE REVIEW

**Comment by clintonb** [comment]
FYI @fredsmith @jibsheet

**Comment by clintonb** [comment]
@feanil any feedback here?

File: playbooks/roles/demo/defaults/main.yml [hunk 1]
```diff
@@ -33,3 +33,7 @@ demo_edxapp_user: 'edxapp'
 demo_edxapp_venv_bin: '{{ COMMON_APP_DIR }}/{{ demo_edxapp_user }}/venvs/{{demo_edxapp_user}}/bin'
 demo_edxapp_course_data_dir: '{{ COMMON_DATA_DIR }}/{{ demo_edxapp_user }}/data'
 demo_edxapp_code_dir: '{{ COMMON_APP_DIR }}/{{ demo_edxapp_user }}/edx-platform'
+
```
Comment (feanil) [inline playbooks/roles/demo/defaults/main.yml]: Why is this in the demo role?   It should be in the local_dev role.

File: playbooks/roles/demo/tasks/main.yml [hunk 2]
```diff
@@ -36,3 +36,18 @@
     owner="{{ demo_edxapp_user }}" group="{{ common_web_group }}"
 
 - include: deploy.yml tags=deploy
+
+- name: create OAuth2 Clients
```
Comment (feanil) [inline playbooks/roles/demo/tasks/main.yml]: same here.  Seems to be in the wrong role.

File: playbooks/roles/launch_ec2/tasks/main.yml [hunk 3]
```diff
@@ -78,6 +78,18 @@
     value: "{{ item.public_dns_name }}"
   with_items: ec2.instances
 
+- name: Add DNS name ecommerce
```
Comment (feanil) [inline playbooks/roles/launch_ec2/tasks/main.yml]: @arbabnazar this DNS name should also be added to the list of removed DNS names for when sandboxes are cleaned up.

Comment (arbabnazar) [inline playbooks/roles/launch_ec2/tasks/main.yml]: @feanil current script will also cover it as it is looking for all the dns records that are associated with the sandbox public dns.

Comment (feanil) [inline playbooks/roles/launch_ec2/tasks/main.yml]: @arbabnazar A sandbox can have more than one public DNS name.  this adds a new one ecommerce-<dns_name>...  So we need to make sure we remove it.

File: playbooks/roles/launch_ec2/tasks/main.yml [hunk 4]
```diff
@@ -78,6 +78,18 @@
...
+  local_action:
```
Comment (feanil) [inline playbooks/roles/launch_ec2/tasks/main.yml]: @clintonb this should be done as a 'with_items list and consolidate the tasks for creating DNS names into a single task.

File: playbooks/roles/launch_ec2/tasks/main.yml (hunk 3 above)
Comment (jibsheet) [inline playbooks/roles/launch_ec2/tasks/main.yml]: @feanil the sandbox code removes anything which is a cname to a sandbox's ec2 dns name, which should cover this.  https://github.com/edx-ops/sysadmin/blob/master/jenkins/terminate-sandbox.py#L111

Comment (feanil) [inline playbooks/roles/launch_ec2/tasks/main.yml]: Ah, got it.

File: playbooks/roles/launch_ec2/tasks/main.yml (hunk 4 above)
Comment (clintonb) [inline playbooks/roles/launch_ec2/tasks/main.yml]: Updated.

File: playbooks/roles/demo/defaults/main.yml (hunk 1 above)
Comment (clintonb) [inline playbooks/roles/demo/defaults/main.yml]: Removed.

File: playbooks/roles/demo/tasks/main.yml (hunk 2 above)
Comment (clintonb) [inline playbooks/roles/demo/tasks/main.yml]: Removed.

**Comment by clintonb** [comment]
@feanil please take another look.

File: playbooks/roles/oauth_client_setup/meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---
```
Comment (feanil) [inline playbooks/roles/oauth_client_setup/meta/main.yml]: File can be deleted.

**Comment by feanil** [comment]
@clintonb one nit but otherwise :+1:

File: playbooks/roles/oauth_client_setup/meta/main.yml (hunk 5 above)
Comment (clintonb) [inline playbooks/roles/oauth_client_setup/meta/main.yml]: Removed.

File: playbooks/edx_sandbox.yml [hunk 6]
```diff
@@ -19,6 +19,8 @@
     COMMON_ENABLE_DATADOG: False
     COMMON_ENABLE_SPLUNKFORWARDER: False
     ENABLE_LEGACY_ORA: !!null
+    ECOMMERCE_NGINX_PORT: '80'
```
Comment (clintonb) [inline playbooks/edx_sandbox.yml]: @feanil any idea why these overrides are not showing up in the ecommerce Nginx file?

**Comment by clintonb** [comment]
@feanil I confirmed that building a sandbox from scratch includes the ecommerce service. Building from the existing AMI does not include ecommerce; but, CMS and LMS work as expected. Ready for a new AMI!!!

**Pull request merged**